examples/src/main/java/org/bitcoinj/examples/ForwardingService.java

```diff
@@ -70,34 +72,54 @@ public static void forward(BitcoinNetwork network, Address address) {
         System.out.println("Forwarding address: " + address);
 
         // Create the Service (and WalletKit)
-        ForwardingService forwardingService = new ForwardingService(address, network);
+        ForwardingService forwardingService = new ForwardingService(network);
 
         // Start the Service (and WalletKit)
         forwardingService.start();
 
         // Start listening and forwarding
-        forwardingService.waitForCoins();
+        final WalletCoinsReceivedEventListener listener = (w, incomingTx, prevBalance, newBalance) -> {
```

Comment:
We could move this to a method and then use a method reference in `addCoinsReceivedEventListener`

Reply:
The method would also align with the `waitForCoins()` method in the diff which would be nice.

Reply:
Sounds good. I'll leave it up to you.